onnx/defs/schema.cc

```diff
@@ -362,14 +362,83 @@ OpSchema& OpSchema::Attr(const Attribute& attr) {
+    }                                                           \
+
+#define ATTR_SETTER_WITH_LIST_VALUE(type, field, attr_type)     \
+    OpSchema& OpSchema::Attr(const std::string& name,           \
+            const std::string& description,                     \
+            const std::vector<type>& default_value)             \
+    {                                                           \
+        AttributeProto a;                                       \
+        a.set_name(name);                                       \
+        a.set_type(attr_type);                                  \
+        for (const auto& v : default_value)                     \
+        {                                                       \
+            a.add_##field(v);                                   \
+        }                                                       \
+        Attr(Attribute(name, description, a));                  \
+        return *this;                                           \
+    }                                                           \
+
+#define ATTR_SETTER_WITH_SINGLE_COMPLEXVALUE(type, field, attr_type)   \
+    OpSchema& OpSchema::Attr(const std::string& name,           \
+            const std::string& description,                     \
+            const type& default_value)                          \
+    {                                                           \
+        AttributeProto a;                                       \
+        a.set_name(name);                                       \
+        *(a.mutable_##field()) = default_value;                 \
+        a.set_type(attr_type);                                  \
+        Attr(Attribute(name, description, a));                  \
+        return *this;                                           \
+    }                                                           \
+
+#define ATTR_SETTER_WITH_LIST_COMPLEXVALUE(type, field, attr_type)     \
+    OpSchema& OpSchema::Attr(const std::string& name,           \
+            const std::string& description,                     \
+            const std::vector<type>& default_value)             \
+    {                                                           \
+        AttributeProto a;                                       \
+        a.set_name(name);                                       \
+        a.set_type(attr_type);                                  \
+        for (const auto& v : default_value)                     \
+        {                                                       \
+            *(a.add_##field()) = v;                             \
+        }                                                       \
+        Attr(Attribute(name, description, a));                  \
+        return *this;                                           \
+    }                                                           \
+
+ATTR_SETTER_WITH_SINGLE_VALUE(int64_t, i, AttributeProto::INT)
```

Comment:
Mix int64_t with AttributeProto::INT is confusing. Similar to other types.

Reply:
These "int64_t", "I", "AttributeProto::INT" are actually redundant information for each other. Because, int64_t should be stored in field "I" and it must have the type as INT. However, using a macro with all these information here is to simplify the macro implementation. Make sense?